dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java

```diff
@@ -334,19 +367,24 @@ private OutputStream compress( HttpServletResponse response, String attachment,
 
         if ( Compression.GZIP == compression )
         {
+            fileName = fileName.replace( "." + format + ".gzip", "" );
+            fileName = fileName.replace( "." + format + ".gz", "" );
+
             response.setHeader( ContextUtils.HEADER_CONTENT_DISPOSITION,
-                "attachment; filename=" + fileName + "." + format + ".gzip" );
+                "attachment; filename=" + fileName + "." + format + ".gz" );
             response.setHeader( ContextUtils.HEADER_CONTENT_TRANSFER_ENCODING, "binary" );
             return new GZIPOutputStream( response.getOutputStream() );
         }
         else if ( Compression.ZIP == compression )
         {
+            fileName = fileName.replace( "." + format + ".zip", "" );
```

Comment:
Could we move this particular operation to a util method with Javadoc and a test? Just to improve test coverage and to make the code more readable/make it easier to understand what is going on.